From f6a5796cccd2daf7a7d3d81dcede0375ff3412ca Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Mon, 7 Apr 2014 00:34:19 -0500
Subject: [PATCH 466/470] rtl8192du: Change to a single copy of a broadcast
 address

Every routine that needed an array of length ETH_ALEN to check
for broadcast addresses had its own private copy. This change converts
to a single global array.

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 core/rtw_ap.c        | 3 +--
 core/rtw_mlme_ext.c  | 4 ++--
 core/rtw_sta_mgt.c   | 5 +----
 include/mlme_osdep.h | 2 ++
 4 files changed, 6 insertions(+), 8 deletions(-)

--- a/drivers/staging/rtl8192du/core/rtw_ap.c
+++ b/drivers/staging/rtl8192du/core/rtw_ap.c
@@ -19,6 +19,7 @@
 #include <osdep_service.h>
 #include <drv_types.h>
 #include <wifi.h>
+#include <mlme_osdep.h>
 
 #ifdef CONFIG_92D_AP_MODE
 
@@ -1940,7 +1941,6 @@ int rtw_ap_inform_ch_switch(struct rtw_a
 	struct sta_priv *pstapriv = &padapter->stapriv;
 	struct mlme_ext_priv *pmlmeext = &padapter->mlmeextpriv;
 	struct mlme_ext_info	*pmlmeinfo = &(pmlmeext->mlmext_info);
-	u8 bc_addr[ETH_ALEN] = {0xff, 0xff, 0xff, 0xff, 0xff, 0xff};
 
 	if ((pmlmeinfo->state&0x03) != WIFI_FW_AP_STATE)
 		return ret;
@@ -1977,7 +1977,6 @@ int rtw_sta_flush(struct rtw_adapter *pa
 	struct sta_priv *pstapriv = &padapter->stapriv;
 	struct mlme_ext_priv *pmlmeext = &padapter->mlmeextpriv;
 	struct mlme_ext_info	*pmlmeinfo = &(pmlmeext->mlmext_info);
-	u8 bc_addr[ETH_ALEN] = {0xff, 0xff, 0xff, 0xff, 0xff, 0xff};
 
 	DBG_8192D(FUNC_NDEV_FMT"\n", FUNC_NDEV_ARG(padapter->pnetdev));
 
--- a/drivers/staging/rtl8192du/core/rtw_mlme_ext.c
+++ b/drivers/staging/rtl8192du/core/rtw_mlme_ext.c
@@ -24,6 +24,8 @@
 #include <mlme_osdep.h>
 #include <recv_osdep.h>
 
+u8 bc_addr[ETH_ALEN] = { 0xff, 0xff, 0xff, 0xff, 0xff, 0xff };
+
 static struct mlme_handler mlme_sta_tbl[] = {
 	{WIFI_ASSOCREQ, "OnAssocReq", &OnAssocReq},
 	{WIFI_ASSOCRSP, "OnAssocRsp", &OnAssocRsp},
@@ -497,7 +499,6 @@ static void _mgt_dispatcher(struct rtw_a
 			    struct mlme_handler *ptable,
 			    struct recv_frame_hdr *precv_frame)
 {
-	u8 bc_addr[ETH_ALEN] = { 0xff, 0xff, 0xff, 0xff, 0xff, 0xff };
 	u8 *pframe = precv_frame->rx_data;
 
 	if (ptable->func) {
@@ -516,7 +517,6 @@ void mgt_dispatcher(struct rtw_adapter *
 #ifdef CONFIG_92D_AP_MODE
 	struct mlme_priv *pmlmepriv = &adapt->mlmepriv;
 #endif /* CONFIG_92D_AP_MODE */
-	u8 bc_addr[ETH_ALEN] = { 0xff, 0xff, 0xff, 0xff, 0xff, 0xff };
 	u8 *pframe = precv_frame->rx_data;
 	struct sta_info *psta =
 	    rtw_get_stainfo(&adapt->stapriv, GetAddr2Ptr(pframe));
--- a/drivers/staging/rtl8192du/core/rtw_sta_mgt.c
+++ b/drivers/staging/rtl8192du/core/rtw_sta_mgt.c
@@ -483,7 +483,6 @@ struct sta_info *rtw_get_stainfo(struct
 	struct sta_info *psta = NULL;
 	u32	index;
 	u8 *addr;
-	u8 bc_addr[ETH_ALEN] = {0xff, 0xff, 0xff, 0xff, 0xff, 0xff};
 
 	if (hwaddr == NULL)
 		return NULL;
@@ -545,10 +544,8 @@ struct sta_info *rtw_get_bcmc_stainfo(st
 {
 	struct sta_info		*psta;
 	struct sta_priv		*pstapriv = &padapter->stapriv;
-	u8 bc_addr[ETH_ALEN] = {0xff, 0xff, 0xff, 0xff, 0xff, 0xff};
-
-	 psta = rtw_get_stainfo(pstapriv, bc_addr);
 
+	psta = rtw_get_stainfo(pstapriv, bc_addr);
 	return psta;
 }
 
--- a/drivers/staging/rtl8192du/include/mlme_osdep.h
+++ b/drivers/staging/rtl8192du/include/mlme_osdep.h
@@ -20,6 +20,8 @@
 #include <osdep_service.h>
 #include <drv_types.h>
 
+extern u8 bc_addr[ETH_ALEN];
+
 void rtw_init_mlme_timer(struct rtw_adapter *padapter);
 void rtw_os_indicate_disconnect(struct rtw_adapter *adapter);
 void rtw_os_indicate_connect(struct rtw_adapter *adapter);
